src/Serilog/Capturing/PropertyValueConverter.cs

```diff
@@ -165,7 +175,15 @@ LogEventPropertyValue CreatePropertyValue(object? value, Destructuring destructu
         if (TryConvertCompilerGeneratedType(value, type, destructuring, out var compilerGeneratedResult))
             return compilerGeneratedResult;
 
-        return new ScalarValue(value.ToString() ?? "");
+        var stringified = value.ToString();
+
+        if (stringified == null)
+            return ScalarValue.EmptyString; // ScalarValue.Null; ?
+
+        if (stringified == string.Empty)
```

Comment:
A similar check is already happening at line 138. IMO we should move the string checks to the bottom and consolidate to something like `var stringified = value is string ? (string)value : value.ToString()`.

Reply:
Similar but not the same. Check on line 138 serves as fast return **before** calling `value.ToString()`.